### PR TITLE
updated abstractSource.php to fix 24698

### PR DESCRIPTION
**Pull request by arushibansal013**
getOptionId() should return id/value on basis of label, as of now returning value as well on basis of value only.
This is causing mismatch in some scenario.

### Fixed Issues (if relevant)

1. magento/magento2#24698: getOptionId() - parameter input is not specific to option value or option label

### Manual testing scenarios (*)

Have any select type attribute of product with options in below mannner
Have an option with label as "09" and suppose its value assigned by magento is "60".
Now later another option has label "60" whose value assigned is any value by magento.
Now Try to get the option id using getOptionId() method in \module-eav\Model\Entity\Attribute\Source\AbstractSource.php
Expected result : The getOptionId should return optionId or value when we pass option label as parameter.
### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by kanthy-i95dev — APPROVED** [review]

**Comment by arushibansal013** [comment]
@sidolov Can u help me out in letting me know how can i change semantic version for an EAV module, as of now no version exists.

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi @arushibansal013. Thanks for collaboration. Can you please resolve conflicts?

**Comment by gabrieldagama** [comment]
The risk was set to`medium` due to: the suggested changes may impact several modules/areas.

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by sidolov** [comment]
@arushibansal013 could you please take a look at failed tests?

File: app/code/Magento/Eav/Model/Entity/Attribute/Source/AbstractSource.php
Comment (engcom-Foxtrot) [inline app/code/Magento/Eav/Model/Entity/Attribute/Source/AbstractSource.php:105]: @arushibansal013 do we really need this method? As it takes $value and returns $value or null.

**Review by engcom-Foxtrot — CHANGES_REQUESTED** [review]

**Comment by engcom-Foxtrot** [comment]
@arushibansal013 I'm closing this pull request due to inactivity. Feel free to reopen.

**Comment by leissbua** [comment]
Crazy a fix was submitted 3 years ago and the maintainers dont merge it, fooling other EE and CE clients. Made us spend some hours to fix what has been fixed yet, but ignored from the maintainers. 

https://github.com/magento/magento2/pull/24935/commits/33e8fe00601e2e0d08ca324259c7076e9797448e

**Pull request closed without merging**
